packages/openapi-generator/src/generator.ts

```diff
@@ -262,19 +263,19 @@ async function generateService(
  * @internal
  */
 export async function getInputFilePaths(input: string): Promise<string[]> {
-  if ((await lstat(input)).isFile()) {
-    return [input];
+  if (glob.hasMagic(input)) {
+    return glob
+      .sync(resolve(input))
+      .filter(path => /(.json|.JSON|.yaml|.YAML|.yml|.YML)$/.test(path));
   }
 
-  const directoryContents = await readdir(input);
+  if (lstatSync(input).isDirectory()) {
+    return glob.sync(
+      resolve(input, './**/*(*.json|*.JSON|*.yaml|*.YAML|*.yml|*.YML)')
+    );
```

Comment:
[req] We should avoid synchronous functions alltogether. I know that we already use this in some places, but still, I think we should rather make this asynchronous. Unfortunately the library still uses callbacks, so we have to put the promise in place on our own. (This also applies to the other use of `glob.sync` above.)

Example:
```ts
return Promise((resolve) => {
  glob(..., (paths) => {
    resolve(paths);
  };
});
```